### PR TITLE
bug: optimize installApp

### PR DESCRIPTION
**Pull request by lingdie**
optimize 
- save write back file: fix codes
- install new image: add log and fix image pull error does not return

### CODE REVIEW

File: pkg/apply/processor/install.go [hunk 1]
```diff
@@ -117,7 +100,28 @@ func (c *InstallProcessor) SyncStatusAndCheck(cluster *v2.Cluster) error {
...
+	}
+
+	prompt := fmt.Sprintf("are you sure to override these following apps? \n%s\t", strings.Join(c.imagesToOverride, "\n"))
+	cancel := "you have canceled to override these apps!"
+	pass, err := confirm.Confirm(prompt, cancel)
+	if err != nil {
+		return err
+	}
+	if !pass {
+		return errors.New(cancel)
```
Comment (fengxsong) [inline pkg/apply/processor/install.go:117]: should ignore cancel error?

File: pkg/apply/applydrivers/apply_drivers_default.go [hunk 2]
```diff
@@ -92,6 +95,11 @@ func (c *Applier) Apply() error {
 		c.ClusterDesired.CreationTimestamp = c.ClusterCurrent.CreationTimestamp
 	}
 	c.updateStatus(clusterErr, appErr)
```
Comment (fengxsong) [inline pkg/apply/applydrivers/apply_drivers_default.go:97]: `appErr` is not used in `updateStatus`

File: pkg/apply/processor/install.go (hunk 1 above)
Comment (lingdie) [reply pkg/apply/processor/install.go:117]: Only moved code place. Do you mean we should ignore appErr if it's cancel?

File: pkg/apply/applydrivers/apply_drivers_default.go (hunk 2 above)
Comment (lingdie) [reply pkg/apply/applydrivers/apply_drivers_default.go:97]: I will change func updateStatus and will use appErr to generate cmdPhase for each cmd.

File: pkg/apply/processor/install.go (hunk 1 above)
Comment (fengxsong) [reply pkg/apply/processor/install.go:117]: > Only moved code place. Do you mean we should ignore appErr if it's cancel?

yes, cancel is not an unexpected error.

Comment (lingdie) [reply pkg/apply/processor/install.go:117]: ok, add a check at return appErr.

**Review by cuisongliu — APPROVED** [review]

**Pull request merged**